cEP-0017.md

```diff
@@ -0,0 +1,366 @@
+to install bears on different distributions by creating a playbook with the required
+installation instructions for all bears.
+
+## Introduction
+
+The current design of the cib tool allows it to perform basic operations such as
+running pip install on bears and then installing its dependencies. However, some
+bears are not even getting installed on certain distributions and cib is running into
+errors. Moreover, `cib` is not able to handle special cases such as of bears
+that have different installation instructions for each distribution
+i.e. `DartLintBear` and bears that have bear dependencies such as the
+`ClangCloneDetectionBear`. Additionally, `cib` needs to be more user friendly
+with a nice interface and should be able to provide verbose output,
+ask user for permission before proceeding.
+
+Another alternative to `cib` that can be used for installing bears and their dependencies
+is Ansible. Ansible allows to orchestrate steps of any process
+for different distributions by creating `playbooks` in YAML.
+Each playbook contains a list of tasks which calls to any of the [ansible modules](http://docs.ansible.com/ansible/latest/modules/modules_by_category.html). A sample playbook
+example is shown below:
+
+```yaml
+tasks:
+  - name: ensure apache is at the latest version
+    yum:
+      name: httpd
+      state: latest
+  - name: write the apache config file
+    template:
+      src: /srv/httpd.j2
+      dest: /etc/httpd.conf
+    notify:
+    - restart apache
+  - name: ensure apache is running (and enable it at boot)
+    service:
+      name: httpd
+      state: started
+      enabled: yes
+```
+
+Ansible provides numerous
+[packaging modules](http://docs.Ansible.com/Ansible/latest/modules/list_of_packaging_modules.html)
+that can be used to install dependencies of `Bears`.
+Another advantage of Ansible is its ability to do a large amount of fact gathering
+from the user's machine and provide relevant information such as `Ansible_distribution` and
+`Ansible_pkg_mgr`. We can use these gathered facts to automatically decide in certain cases
+ such as shown below.
+
+```yaml
+- name: install basic package
+  action: >
+    {{ ansible_pkg_mgr }} name=package_name
+```
+
+Other than this, Ansible also provides options to gain root access for installing packages
+and printing verbose output. Ansible also allows filtering tasks based on tags.
+This can be used to install bears for a specific language, certain capability etc
+if the appropriate tags are added.
+
+## Proposed Changes
+
+### Procedure for the installation of the bears using cib
+
+1. Identify the valid bear names that need to be installed as specified by the
+   user on the command line.
+2. Install bear on the valid bear names. We will run commands from `CondaRequirement`
+   to check whether the package is uploaded to `Anaconda.org` and then check whether
+   it can be installed from `PyPi` using `PipRequirement`.
```

Comment:
-> "PyPi" ; it is an abbreviation, not a string literal.
A link would also be helpful